src/background/listener/bee-api.listener.ts

```diff
@@ -73,6 +108,36 @@ export class BeeApiListener {
     )
   }
 
+  /**
+   *
+   * @param bzzUrl BZZ URL with arbitrary query parameters,  e.g. http://.../1231abcd.../valami.html?swarm-session-id=vmi&smth=5
+   * @returns Swarm Session ID string
+   */
+  private getSwarmSessionIdFromUrl(bzzUrl: string): string | null {
+    const queryIndex = bzzUrl.indexOf('?')
+
+    if (queryIndex === -1) return null
+
+    const invalidQueryEnds = bzzUrl.indexOf('/', queryIndex + 1)
+
+    if (invalidQueryEnds !== -1) {
+      // the swarm session id must be in the invalid query param
+      const invalidQuery = bzzUrl.slice(queryIndex + 1, invalidQueryEnds)
+      const sessionIdSplit = invalidQuery.split(`${SWARM_SESSION_ID_KEY}=`)
+
+      if (sessionIdSplit.length !== 2) {
+        // if there is no swarm session ID in the invalid query
+        // then the request will fail
+        return null
+      }
+
+      return sessionIdSplit[1]
+    }
+    // if the process reaches this point, the given url can be handled as with valid query parameter
+
+    return new URL(bzzUrl).searchParams.get(SWARM_SESSION_ID_KEY)
+  }
```

Comment:
Can you use `URLSearchParams` to easily access and manipulate query string parameters?

Reply:
unfortunately not, because the given URL can be malformatted and the `URLSearchParams` does not handle that scenario.

For example if I initialize a `Bee` object from `bee-js` by passing the fakeUrl with the corresponding `swarm-session-id` as a query parameter, the Bee JS library will append its path (or optionally its query params) to that given address, e.g. `http://localhost:1633/fake-url/bee-api?swarm-session-id=smth/dirs`